src/backend/distributed/operations/shard_split.c

```diff
@@ -0,0 +1,843 @@
+			CreateObjectOnPlacement(splitShardCreationCommandList, workerPlacementNode);
+		}
+	}
+
+	/* Perform Split Copy */
+	DoSplitCopy(sourceShardNode, sourceColocatedShardIntervalList,
+				shardGroupSplitIntervalListList, workersForPlacementList);
+
+	/*
+	 * Create auxillary structures post copy.
+	 */
+	foreach_ptr(shardIntervalList, shardGroupSplitIntervalListList)
+	{
+		/* Iterate on split shard interval list and corresponding placement worker */
+		ShardInterval *shardInterval = NULL;
+		WorkerNode *workerPlacementNode = NULL;
+		forboth_ptr(shardInterval, shardIntervalList, workerPlacementNode,
+					workersForPlacementList)
+		{
+			List *indexCommandList = GetPostLoadTableCreationCommands(
+				shardInterval->relationId,
+				true /* includeIndexes */,
+				true /* includeReplicaIdentity */);
+			indexCommandList = WorkerApplyShardDDLCommandList(
+				indexCommandList,
+				shardInterval->shardId);
+
+			CreateObjectOnPlacement(indexCommandList, workerPlacementNode);
+		}
+	}
+}
+
+
+/*
+ * Perform Split Copy from source shard(s) to split children.
+ * 'sourceShardNode'					: Source shard worker node.
+ * 'sourceColocatedShardIntervalList'	: List of source shard intervals from shard group.
+ * 'shardGroupSplitIntervalListList'	: List of shard intervals for split children.
+ * 'workersForPlacementList'			: List of workers for split children placement.
+ */
+static void
+DoSplitCopy(WorkerNode *sourceShardNode, List *sourceColocatedShardIntervalList,
+			List *shardGroupSplitIntervalListList, List *destinationWorkerNodesList)
+{
+	ShardInterval *sourceShardIntervalToCopy = NULL;
+	List *splitShardIntervalList = NULL;
+
+	int taskId = 0;
+	List *splitCopyTaskList = NULL;
+	forboth_ptr(sourceShardIntervalToCopy, sourceColocatedShardIntervalList,
+				splitShardIntervalList, shardGroupSplitIntervalListList)
+	{
+		StringInfo splitCopyUdfCommand = CreateSplitCopyCommand(sourceShardIntervalToCopy,
+																splitShardIntervalList,
+																destinationWorkerNodesList);
+
+		Task *splitCopyTask = CreateBasicTask(
+			sourceShardIntervalToCopy->shardId, /* jobId */
+			taskId,
+			READ_TASK,
+			splitCopyUdfCommand->data);
+
+		ShardPlacement *taskPlacement = CitusMakeNode(ShardPlacement);
+		SetPlacementNodeMetadata(taskPlacement, sourceShardNode);
+
+		splitCopyTask->taskPlacementList = list_make1(taskPlacement);
+
+		splitCopyTaskList = lappend(splitCopyTaskList, splitCopyTask);
+		taskId++;
+	}
+
+	ExecuteTaskListOutsideTransaction(ROW_MODIFY_NONE, splitCopyTaskList,
+									  MaxAdaptiveExecutorPoolSize,
+									  NULL /* jobIdList (ignored by API implementation) */);
+}
+
+
+/*
+ * Create Copy command for a given shard source shard to be copied to corresponding split children.
+ * 'sourceShardSplitInterval' : Source shard interval to be copied.
+ * 'splitChildrenShardINnerIntervalList' : List of shard intervals for split children.
+ * 'destinationWorkerNodesList' : List of workers for split children placement.
+ */
```

Comment:
would be good to add an example of what the command looks like

Reply:
Ack. Will do.